### DISCUSSION TITLE
Activate language features solely on file open (in theia IDE)

### DISCUSSION THREAD

**Discussion by chbo123456** [eclipse-langium/langium#1369] [Q&A] [answered]
Hi,

is it possible to active the language features only to the file(s) which are currently opened by the IDE? Usually the whole workspace is parsed on extension activation and if the workspace contains many files, this makes the (theia) IDE pretty slow. If yes, can this be made configurable in a "lunch.json" file or so?

Thanks!

**Comment by msujew** [comment]
Hey @chbo123456,

you will effectively need to build a toggle for this behavior:

https://github.com/eclipse-langium/langium/blob/8a535521fd131f5214c5ede26a92e406fbf57f86/packages/langium/src/workspace/workspace-manager.ts#L80-L83

You could maybe create a config format for your language (similar to `tsconfig.json`) and create an option in there that decides whether or not to load the whole workspace. Even better, it could contain a setting that limits the search pattern using something like `glob`. Since the `launch.json` has no influence on runtime behavior of the IDE (it is mainly used for debugging), placing the toggle there would be not advisable.

**Reply by chbo123456** [reply]
@msujew Do you mean to override the traverseFolder method?

**Reply by msujew** [reply, marked as answer]
Either `traverseFolder` or the whole `initializeWorkspace` method if needed. It depends on how much you actually need to modify the behavior.

**Reply by chbo123456** [reply]
Thanks!
